backend/src/permissions.js

```diff
@@ -1,23 +1,35 @@
-import { rule, shield, allow } from "graphql-shield";
+import { rule, shield, allow, deny } from "graphql-shield";
+
+import User from "./db/entities/User";
 
 const isAuthenticated = rule({ cache: "contextual" })(
-  async (_parent, _args, { dataSources, id }) => {
-    let exists = dataSources.db.userExists(id);
+  async (_parent, _args, { id }) => {
+    let exists = await User.exists({ id });
     return exists;
   }
 );
 
 export const permissions = shield(
   {
+    User: allow,
+    Post: allow,
+
     Query: {
-      "*": allow,
```

Comment:
`'*': deny,`
for UserData i like the explicit is better than implicit approach more, also you can find errors in your permissions more easily